packages/tracing/src/browser/browsertracing.ts

```diff
@@ -187,22 +192,27 @@ export class BrowserTracing implements Integration {
     // eslint-disable-next-line @typescript-eslint/unbound-method
     const { beforeNavigate, idleTimeout, maxTransactionDuration } = this.options;
 
-    // if beforeNavigate returns undefined, we should not start a transaction.
+    const parentContextFromHeader = context.op === 'pageload' ? getHeaderContext() : undefined;
+
     const expandedContext = {
       ...context,
-      ...getHeaderContext(),
+      ...parentContextFromHeader,
       trimEnd: true,
     };
     const modifiedContext = typeof beforeNavigate === 'function' ? beforeNavigate(expandedContext) : expandedContext;
 
-    if (modifiedContext === undefined) {
-      logger.log(`[Tracing] Did not create ${context.op} idleTransaction due to beforeNavigate`);
-      return undefined;
+    // TODO (kmclb): for backwards compatibility reasons, beforeNavigate can return undefined to "drop" the transaction
+    // (prevent it from being sent to Sentry). This can be removed in V6, after which time modifiedContext and
+    // finalContext will be one and the same.
+    const finalContext = modifiedContext === undefined ? { ...expandedContext, sampled: false } : modifiedContext;
```

Comment:
L:

So the old behavior here was to abort and not create a transaction. The new behavior is create a transaction and set `sampled: false`.
This definitely changes the previous behavior, but I'm okay with that as long as this was not accidental.

There still won't be any new transaction sent to Sentry, but outgoing requests can propagate the sampling decision.

Alternatively, we could preserve the old behavior, and simply let new uses of `beforeNavigate` use `sampled: false` instead of returning `undefined` (and we need an update in docs to reflect what's the way we intend the hook to be used).

Reply:
> This definitely changes the previous behavior, but I'm okay with that as long as this was not accidental.

You and I had a conversation about this, I thought? Back in August? And agreed that this is what should happen, no?